pkg/operator/configobserver/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,169 @@
+}
+
+// ObserveAccessTokenInactivityTimeout returns an unstructured fragment of KubeAPIServerConfig that has access token inactivity timeout,
+// if there is a valid value for it in OAuth cluster config.
+func ObserveAccessTokenInactivityTimeout(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	errs = []error{}
+	listers := genericlisters.(OAuthLister)
+	observedConfig := map[string]interface{}{}
+	tokenConfigPath := []string{"oauthConfig", "tokenConfig"}
+	accessTokenInactivityTimeoutPath := append(tokenConfigPath, fieldAccessTokenInactivityTimeoutSeconds)
+	defer func() {
+		// Prune the observed config so that it only contains access token inactivity timeout field.
+		ret = configobserver.Pruned(ret, accessTokenInactivityTimeoutPath)
+	}()
+
+	existingAccessTokenInactivityTimeout, _, err := unstructured.NestedFieldCopy(existingConfig, accessTokenInactivityTimeoutPath...)
+	if err != nil {
+		// If failed to read existing config for access token inactivity timeout, continue execution with the
+		// hope to find the value for it in OAuth cluster config.
+		errs = append(errs, err)
+	}
+
+	var existingAccessTokenInactivityTimeoutFloat float64
+	if existingAccessTokenInactivityTimeout != nil {
+		if f, ok := existingAccessTokenInactivityTimeout.(float64); ok {
+			existingAccessTokenInactivityTimeoutFloat = f
+		}
+	}
+
+	oauthConfig, err := listers.OAuthLister().Get("cluster")
+	if err != nil {
+		// Failed to read OAuth cluster config.
+		if errors.IsNotFound(err) {
+			klog.Warning("oauth.config.openshift.io/cluster: not found")
+		}
+
+		if existingAccessTokenInactivityTimeoutFloat == 0 {
+			// As the nil value indicates that an attempt to read from existing config failed previously, return empty config.
+			return buildUnstructuredTokenConfig("", nil, tokenConfigPath), append(errs, err)
+		}
+		// Return whatever is present in existing config
+		return existingConfig, append(errs, err)
+	}
+
+	observedAccessTokenInactivityTimeout := oauthConfig.Spec.TokenConfig.AccessTokenInactivityTimeout.Seconds()
+	if observedAccessTokenInactivityTimeout != 0 {
+		// Build observed config only if the inactivity timeout value is set in OAuth cluster config.
+		observedConfig = buildUnstructuredTokenConfig(fieldAccessTokenInactivityTimeoutSeconds,
+			observedAccessTokenInactivityTimeout, tokenConfigPath)
+	}
+
+	if existingAccessTokenInactivityTimeoutFloat != observedAccessTokenInactivityTimeout {
+		recorder.Eventf("ObserveTokenConfig", "%s changed from %v to %v", fieldAccessTokenInactivityTimeoutSeconds,
+			existingAccessTokenInactivityTimeoutFloat,
+			observedAccessTokenInactivityTimeout)
+	}
+
+	return observedConfig, errs
+}
+
+func buildUnstructuredTokenConfig(key string, val interface{}, fields []string) map[string]interface{} {
+	defaultConfig := map[string]interface{}{}
+
+	observedMaxAgeFieldMap := map[string]interface{}{
```

Comment:
nit: this is not about maxagefield anymore, is it? It's generic.

Reply:
Done.